share/eds/eds_test.go

```diff
@@ -175,6 +176,36 @@ func TestReadEDSContentIntegrityMismatch(t *testing.T) {
 	require.ErrorContains(t, err, "share: content integrity mismatch: imported root")
 }
 
+// BenchmarkReadWriteEDS benchmarks the time it takes to write and read an EDS from disk.
+// The benchmark is run with a 4x4 ODS to a 64x64 ODS - a higher value can be used but it will run for much longer.
+func BenchmarkReadWriteEDS(b *testing.B) {
+	ctx, cancel := context.WithCancel(context.Background())
+	b.Cleanup(cancel)
+	for originalDataWidth := 4; originalDataWidth <= 64; originalDataWidth *= 2 {
+		tmpDir := b.TempDir()
+		err := os.Chdir(tmpDir)
+		require.NoError(b, err)
+		eds := share.RandBenchmarkEDS(b, originalDataWidth)
+		dah := da.NewDataAvailabilityHeader(eds)
+		b.Run(fmt.Sprintf("Writing %dx%d", originalDataWidth, originalDataWidth), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				f, _ := os.OpenFile(fmt.Sprintf("%d.car", originalDataWidth), os.O_WRONLY|os.O_CREATE, 0600)
+				err := WriteEDS(ctx, eds, f)
+				require.NoError(b, err)
+				f.Close()
+			}
+		})
+		b.Run(fmt.Sprintf("Reading %dx%d", originalDataWidth, originalDataWidth), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				f, _ := os.OpenFile(fmt.Sprintf("%d.car", originalDataWidth), os.O_RDONLY, 0600)
```

Comment:
same thing here about Hlib's comment on using an in memory buffer instead of a file to have a cleaner benchmark that isn't touching disk i/o

Reply:
done